src/ophyd_async/epics/eiger/_odin_io.py

```diff
@@ -81,12 +83,23 @@ async def open(self, name: str, exposures_per_event: int = 1) -> dict[str, DataK
             self._drv.file_path.set(str(info.directory_path)),
             self._drv.file_name.set(info.filename),
             self._drv.data_type.set(
-                "uint16"
+                "UInt16"
             ),  # TODO: Get from eiger https://github.com/bluesky/ophyd-async/issues/529
             self._drv.num_to_capture.set(0),
         )
 
-        await self._drv.capture.set(Writing.ON)
+        await wait_for_value(self._drv.meta_active, "Active", timeout=DEFAULT_TIMEOUT)
+
+        await set_and_wait_for_other_value(
```

Comment:
Could: A couple of extra tests here would be good:
* That we wait for `Active` before we trigger the capture
* That we wait for `Writing` before we finish the open

Reply:
We've used `asyncio.Event` to help with testing this kind of stuff in `dodal`, see https://github.com/DiamondLightSource/dodal/blob/5f73fa09fc30e021da63e61e71c05c1378488214/tests/devices/unit_tests/i03/test_undulator_dcm.py#L144